src/interfaces/types.ts

```diff
@@ -6,8 +6,18 @@ type ClassDeclaration = Type;
 
 type IResolvers = Record<string, any>;
 
+type IQueryTypes = 'Query' | 'Mutation' | 'Subscription' | 'Field';
 type IResolverFunction = (parent: unknown, args: any, ctx: any, info: unknown) => unknown;
 type IResolverParamType = 'parent' | 'args' | 'ctx' | 'info';
 type ResolverDeclaration = ClassDeclaration;
+type ISubscriptionFilterFunction = (payload: any, variables: any) => void;
```

Comment:
Maybe this could better fit in a `subscription.decorator.types.ts` file as it only relates to the decorator.